src/core/bonded_interactions/bonded_interactions.dox

```diff
@@ -24,131 +24,164 @@
+ *    pair_energy(Utils::Vector3d const &dx) const;
+ *    @endcode
+ *    This function returns the bond energy. The same information as given for
+ *    the force calculation above applies here. This function will be called
+ *    from @ref energy_inline.hpp .
+ *  * An explicit default contructor, for example
+ *    @code{.cpp}
+ *    Fene_bond_parameters() = default;
+ *    @endcode
+ *  * A custom constructor which is used to set all parameters and to do
+ *    preparatory calculations if necessary, for example
  *    @code{.cpp}
- *    inline boost::optional<Utils::Vector3d>
- *    fene_pair_force(Bonded_ia_parameters const &iaparams,
- *                    Utils::Vector3d const &dx);
- *    inline boost::optional<double>
- *    fene_pair_energy(Bonded_ia_parameters const &iaparams,
- *                     Utils::Vector3d const &dx);
+ *    Fene_bond_parameters(double k, double drmax, double r0);
  *    @endcode
- *  * The setter function gets a \p bond_type which is a numerical id
- *    identifying the number of the bond type in the simulation.
- *    It DOES NOT determine the type of the bond potential (harmonic vs FENE).
- *    The signature of the setter function has to contain the \c bond_type, the
- *    remaining parameters are specific to the interaction:
+ *    All values the bond needs to function properly should be passed as
+ *    arguments to this constructor.
+ *  * A template function for serialization called \c serialize. This is for
+ *    communication between nodes in parallel computations.
+ *    The following function can serve as a starting point.
  *    @code{.cpp}
- *    int fene_set_params(int bond_type, double k, double drmax, double r0);
+ *    template <typename Archive>
+ *    void serialize(Archive &ar, long int /* version */) {
+ *      ar &k;
+ *      ar &drmax;
+ *      ar &r0;
+ *      ar &drmax2;
+ *      ar &drmax2i;
+ *    }
  *    @endcode
- *    @ref ES_OK is returned on success, @ref ES_ERROR on error, e.g. when
- *    parameters are invalid.
- *  * The setter function must call @ref make_bond_type_exist() with
- *    \p bond_type to allocate the memory for storing the parameters.
- *  * Afterwards, the bond parameters can be stored in the global variable
- *    @ref bonded_ia_params "bonded_ia_params[bond_type]"
- *  * \c bonded_ia_params[bond_type].num is the number of particles involved
- *    in the bond, minus 1, i.e. 1 for a pairwise bonded potential such as the
- *    FENE bond.
- *  * The parameters for the individual bonded interaction go to the member of
- *    @ref Bond_parameters for your interaction defined in the previous step.
- *    For the FENE bond, this would be:
+ *    Replace all the <tt>ar &</tt> commands with the new bond's parameters.
+ *    Every data member of your struct needs to be transmitted. This template
+ *    function can be made private, but then the following line needs to be
+ *    added before for the serialization routine to be able to access it:
```

Comment:
serialization methods are typically private, therefore I would rephrase it like so:
```suggestion
 *    Every data member of your struct needs to be transmitted. This template
 *    function is private, hence the following line needs to be added before:
```